probe/endpoint/conntrack.go

```diff
@@ -14,87 +13,92 @@ import (
-	Destroy          = "destroy"
+	xmlHeader        = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"
+	conntrackOpenTag = "<conntrack>\n"
+	timeWait         = "TIME_WAIT"
+	tcpProto         = "tcp"
+	newType          = "new"
+	updateType       = "update"
+	destroyType      = "destroy"
 )
 
-// Layer3 - these structs are for the parsed conntrack output
-type Layer3 struct {
+type layer3 struct {
 	XMLName xml.Name `xml:"layer3"`
 	SrcIP   string   `xml:"src"`
 	DstIP   string   `xml:"dst"`
 }
 
-// Layer4 - these structs are for the parsed conntrack output
-type Layer4 struct {
+type layer4 struct {
 	XMLName xml.Name `xml:"layer4"`
 	SrcPort int      `xml:"sport"`
 	DstPort int      `xml:"dport"`
 	Proto   string   `xml:"protoname,attr"`
 }
 
-// Meta - these structs are for the parsed conntrack output
-type Meta struct {
+type meta struct {
 	XMLName   xml.Name `xml:"meta"`
 	Direction string   `xml:"direction,attr"`
-	Layer3    Layer3   `xml:"layer3"`
-	Layer4    Layer4   `xml:"layer4"`
+	Layer3    layer3   `xml:"layer3"`
+	Layer4    layer4   `xml:"layer4"`
 	ID        int64    `xml:"id"`
 	State     string   `xml:"state"`
 }
 
-// Flow - these structs are for the parsed conntrack output
-type Flow struct {
+type flow struct {
 	XMLName xml.Name `xml:"flow"`
-	Metas   []Meta   `xml:"meta"`
+	Metas   []meta   `xml:"meta"`
 	Type    string   `xml:"type,attr"`
 
-	Original, Reply, Independent *Meta `xml:"-"`
+	Original, Reply, Independent *meta `xml:"-"`
 }
 
 type conntrack struct {
 	XMLName xml.Name `xml:"conntrack"`
-	Flows   []Flow   `xml:"flow"`
+	Flows   []flow   `xml:"flow"`
 }
 
-// Conntracker is something that tracks connections.
-type Conntracker interface {
-	WalkFlows(f func(Flow))
-	Stop()
+// flowWalker is something that maintains flows, and provides an accessor
+// method to walk them.
+type flowWalker interface {
+	walkFlows(f func(flow))
+	stop()
 }
 
-// Conntracker uses the conntrack command to track network connections
-type conntracker struct {
+type nilFlowWalker struct{}
+
+func (n *nilFlowWalker) stop()                  {}
+func (n *nilFlowWalker) walkFlows(f func(flow)) {}
+
+// conntrackWalker uses the conntrack command to track network connections and
+// implement flowWalker.
+type conntrackWalker struct {
 	sync.Mutex
 	cmd           exec.Cmd
-	activeFlows   map[int64]Flow // active flows in state != TIME_WAIT
-	bufferedFlows []Flow         // flows coming out of activeFlows spend 1 walk cycle here
+	activeFlows   map[int64]flow // active flows in state != TIME_WAIT
+	bufferedFlows []flow         // flows coming out of activeFlows spend 1 walk cycle here
 	existingConns bool
 	args          []string
 	quit          chan struct{}
 }
 
-// NewConntracker creates and starts a new Conntracter
-func NewConntracker(existingConns bool, args ...string) (Conntracker, error) {
+// newConntracker creates and starts a new conntracker.
+func newConntrackFlowWalker(useConntrack, existingConns bool, args ...string) flowWalker {
```

Comment:
We could extend the tests to deal with that behaviour.  But thats work....

Oh go on then I'll do it.